recipes/eleuther_eval.py

```diff
@@ -4,40 +4,298 @@
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
 
+import importlib.metadata
 import sys
 import time
 
-from typing import Any, Dict, List, Tuple, Union
+from typing import Dict, List, Tuple, Union
+
+import PIL
 
 import torch
 from omegaconf import DictConfig
 
-from torch import nn
-from torchtune import config, generation, training, utils
-from torchtune.data import left_pad_sequence
+from torchtune import config, training, utils
+from torchtune.data import (
+    format_content_with_images,
+    left_pad_sequence,
+    Message,
+    padded_collate_tiled_images_and_mask,
+)
+from torchtune.generation import generate, sample
 from torchtune.modules import TransformerDecoder
+from torchtune.modules.model_fusion import DeepFusionModel
 from torchtune.modules.tokenizers import ModelTokenizer
+from torchtune.modules.transforms import Transform
 from torchtune.recipe_interfaces import EvalRecipeInterface
 
-logger = utils.get_logger("DEBUG")
-
 try:
     import lm_eval
-    from lm_eval.evaluator import evaluate, get_task_list
-    from lm_eval.models.huggingface import HFLM
-    from lm_eval.tasks import get_task_dict, TaskManager
-    from lm_eval.utils import make_table
 except ImportError:
-    logger.error(
-        "Recipe requires EleutherAI Eval Harness v0.4. Please install with `pip install lm_eval==0.4.*`"
+    print(
+        "You must install the EleutherAI Eval Harness to run this recipe. "
+        "Please install with `pip install lm_eval>=0.4.4`"
+    )
+    sys.exit(1)
+
+lm_eval_version = importlib.metadata.version("lm_eval")
+if not lm_eval_version >= "0.4.2":
+    print(
+        "You must install the EleutherAI Eval Harness >= v0.4.2 to run this recipe. "
+        "Please install with `pip install lm_eval>=0.4.4`"
```

Comment:
Why not raise? Also version numbers don't match

Reply:
Hard to catch raises in the tests. 

I want to investigate this further, but it's a follow-up PR.

Reply:
OK fair enough. The version numbers though? What's with 0.4.2/0.4.4 mismatch

Reply:
Fixed.